src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyNavigationManager.cs

```diff
@@ -24,13 +24,12 @@ internal WebAssemblyNavigationManager()
         {
         }
 
-        protected override void EnsureInitialized()
+        public WebAssemblyNavigationManager(WebAssemblyJSRuntimeInvoker jSRuntimeInvoker)
         {
-            // As described in the comment block above, BrowserNavigationManager is only for
-            // client-side (Mono) use, so it's OK to rely on synchronicity here.
-            var baseUri = DefaultWebAssemblyJSRuntime.Instance.Invoke<string>(Interop.GetBaseUri);
-            var uri = DefaultWebAssemblyJSRuntime.Instance.Invoke<string>(Interop.GetLocationHref);
+            var baseUri = jSRuntimeInvoker.InvokeUnmarshalled<object, object, object, string>(Interop.GetBaseUri, null, null, null);
```

Comment:
You'd have to change the typescript code to return a .NET string. Something like this: https://github.com/dotnet/aspnetcore/blob/blazor-wasm/src/Components/Web.JS/src/Platform/WebAssemblyConfigLoader.ts#L6

Reply:
Aha! Today I learned!